src/eradiate/scenes/surface/_dem.py

```diff
@@ -0,0 +1,277 @@
+        init_type=".BufferMeshShape or .OBJMeshShape or .PLYMeshShape or dict",
+    )
+
+    bsdf: BSDF = documented(
+        attr.ib(
+            factory=LambertianBSDF,
+            converter=bsdf_factory.convert,
+            validator=attr.validators.instance_of(BSDF),
+        ),
+        doc="The reflection model attached to the surface.",
+        type=".BSDF",
+        init_type=".BSDF or dict, optional",
+        default=":class:`LambertianBSDF() <.LambertianBSDF>`",
+    )
+
+    @staticmethod
+    def _vertex_index(x, y, len_y):
+        return x * len_y + y
+
+    @staticmethod
+    def _generate_face_indices(len_x, len_y):
+        face_indices = []
+        for x in range(len_x + 1):
+            for y in range(len_y + 1):
+                vertex_sw = DEMSurface._vertex_index(x, y, len_y + 2)
+                vertex_ne = DEMSurface._vertex_index(x + 1, y + 1, len_y + 2)
+                vertex_nw = DEMSurface._vertex_index(x, y + 1, len_y + 2)
+                vertex_se = DEMSurface._vertex_index(x + 1, y, len_y + 2)
+                # In mitsuba, triangles are defined clockwise
+                face_indices.append((vertex_sw, vertex_ne, vertex_nw))
+                face_indices.append((vertex_sw, vertex_se, vertex_ne))
+
+        return face_indices
+
+    @classmethod
+    def from_dataarray(
+        cls, data: xr.DataArray, bsdf: BSDF, planet_radius=6371 * ureg.km
+    ) -> DEMSurface:
+        """
+        This classmethod enables the instantiation of a DEM from an xarray data array holding elevation data.
+        The dataarray must use latitude and longitude as its dimensional coordinates.
+
+        Parameters
+        ----------
+        data : DataArray
+            Dataarray holding the elevation data. Dimensional coordinates must be latitude `lat`
+            and longitude `lon`. Data will be interpreted to be given in kernel units.
+
+        bsdf : .BSDF
+            BSDF to be attached to the mesh shape.
+
+        planet_radius : quantity
+            Planet radius used to convert latitude and longitude into distance units.
+            Defaults to Earth's radius.
+
+        Returns
+        -------
+        :class:`.DEMSurface`
+            Created :class:`DEMSurface`.
+        """
+        radius = planet_radius.to(ucc.get("length"))
+
+        len_lat = len(data.lat)  # x
+        len_lon = len(data.lon)  # y
+
+        # I convert the degrees to local distances based on the planet radius
+        # This will be handled differently when I include the AtmosphereGeometry.
+        lon_quantity = to_quantity(data.lon)
+        lat_quantity = to_quantity(data.lat)
+        mean_lon = lon_quantity.mean().to(ureg.rad)
+        lon_range = lon_quantity.max() - lon_quantity.min()
+        lat_range = lat_quantity.max() - lat_quantity.min()
+        lon_length = (lon_range.to(ureg.rad) * radius).magnitude
+        lat_length = (lat_range.to(ureg.rad) * radius * np.cos(mean_lon)).magnitude
```

Comment:
For performance, let's remove units ASAP (I didn't check how this impacts the rest of the code, please verify):

```suggestion
        lon_rad = to_quantity(data.lon).m_as(ureg.rad)
        lat_rad = to_quantity(data.lat).m_as(ureg.rad)
        mean_lon = lon_rad.mean()
        lon_range = lon_rad.max() - lon_rad.min()
        lat_range = lat_rad.max() - lat_rad.min()
        lon_length = lon_range * radius
        lat_length = lat_range * radius * np.cos(mean_lon)
```